contracts/peg/SafePSM.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.8.4;
+
+import "./PriceBoundPSM.sol";
+
+contract SafePSM is PriceBoundPSM {
+
+    /// @notice the oracle reference by the contract
+    IOracle public actualPriceOracle;
+
+    constructor(
+        uint256 _floor,
+        uint256 _ceiling,
+        OracleParams memory _params,
+        uint256 _mintFeeBasisPoints,
+        uint256 _redeemFeeBasisPoints,
+        uint256 _reservesThreshold,
+        uint256 _feiLimitPerSecond,
+        uint256 _mintingBufferCap,
+        IERC20 _underlyingToken,
+        IPCVDeposit _surplusTarget,
+        IOracle _actualPriceOracle
+    ) PriceBoundPSM(
+        _floor,
+        _ceiling,
+        _params,
+        _mintFeeBasisPoints,
+        _redeemFeeBasisPoints,
+        _reservesThreshold,
+        _feiLimitPerSecond,
+        _mintingBufferCap,
+        _underlyingToken,
+        _surplusTarget
+    ) {
+        actualPriceOracle = _actualPriceOracle;
+    }
+
+    /// @notice the price of the referenced oracle
+    /// @return the peg as a Decimal
+    /// @dev the peg is defined as FEI per X with X being ETH, dollars, etc
+    function readActualOracle() public view returns (Decimal.D256 memory) {
```

Comment:
I think you should use the normal oracle & instead override _getMintAmountOut and _getRedeemAmountOut to not read the oracle value (that will be fixed to 1 anyway)
you don't need the oracle that returns 1
and then you can keep the default behavior on _validatePriceRange & _validPrice
```
function _getMintAmountOut(uint256 amountIn) private view returns (uint256 amountFeiOut) {
        amountFeiOut = amountIn
            .mul(Constants.BASIS_POINTS_GRANULARITY - mintFeeBasisPoints)
            .div(Constants.BASIS_POINTS_GRANULARITY)
            .asUint256();
    }
```
something like that

Reply:
will address this in a separate pr